homeassistant/components/sensor/daikin.py

```diff
@@ -0,0 +1,122 @@
+"""
+Support for Daikin AC Sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_ICON,
+    CONF_MONITORED_CONDITIONS, CONF_TEMPERATURE_UNIT
+)
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.daikin import (
+    SENSOR_TYPES,
+    ATTR_INSIDE_TEMPERATURE, ATTR_OUTSIDE_TEMPERATURE,
+    daikin_api_setup
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSOR_TYPES.keys()):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Daikin sensors."""
+    if discovery_info is not None:
+        host = discovery_info.get('ip')
+        name = None
+        monitored_conditions = discovery_info.get(
+            CONF_MONITORED_CONDITIONS, list(SENSOR_TYPES.keys())
+        )
+        _LOGGER.info("Discovered a Daikin AC sensor on %s", host)
+    else:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        monitored_conditions = config.get(CONF_MONITORED_CONDITIONS)
+        _LOGGER.info("Added Daikin AC sensor on %s", host)
+
+    api = daikin_api_setup(hass, host, name)
+
+    sensors = []
+    for monitored_state in monitored_conditions:
+        sensors.append(DaikinClimateSensor(api, name, monitored_state))
+
+    add_devices(sensors, True)
+
+
+class DaikinClimateSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, api, name=None, monitored_state=SENSOR_TYPES.keys()):
```

Comment:
The default value for the keyword argument seems to never be used. I'd exchange the keyword argument for a regular positional argument. Also, it's error prone to default keyword arguments to mutable objects.